gen/struct_test.go

```diff
@@ -992,10 +1011,13 @@ func TestStructJSON(t *testing.T) {
 			assert.Equal(t, tt.j, string(encoded))
 		}
 
-		v := reflect.New(reflect.TypeOf(tt.v).Elem()).Interface()
+		u := tt.v
+		if tt.w != nil {
+			u = tt.w
+		}
+		v := reflect.New(reflect.TypeOf(u).Elem()).Interface()
```

Comment:
Also might be better to use full names here, since they are drifting from being abbreviations of meaningful names. `v` in scope no longer means value, but “type of value”. `w` appears to be an alternate value for purposes of inferring the type from a more complete object.

This would probably all be a lot more clear with the three relevant cases extracted to a clearer test with meaningful names.